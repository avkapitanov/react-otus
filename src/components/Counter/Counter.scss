.react-hw04__sw-counter {
  font-weight: 700;
  margin-bottom: 20px;
  width: fit-content;
}
