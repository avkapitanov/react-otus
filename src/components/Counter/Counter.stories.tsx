import React from 'react';
import { Counter, CounterProps } from './Counter';

import { Story, Meta } from '@storybook/react';

export default {
  title: 'Homework/Counter',
  component: Counter,
} as Meta;

const Template: Story<CounterProps> = (args) => <Counter {...args} />;

export const Empty = Template.bind({});
Empty.args = {
  loadedPeople: 0,
};

export const WithLoaded = Template.bind({});
WithLoaded.args = {
  loadedPeople: 5,
};
