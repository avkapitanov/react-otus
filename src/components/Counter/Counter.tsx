import React, { Component } from 'react';
import './Counter.scss';

type CounterProps = {
  loadedPeople: number;
};

class Counter extends Component<CounterProps> {
  constructor(props: CounterProps) {
    super(props);
  }

  render() {
    const { loadedPeople } = this.props;

    if (loadedPeople == 0) {
      return (
        <div
          className="react-hw04__sw-counter react-hw04__sw-counter--empty"
          data-testid="empty-counter"
        >
          Пока не загружено ни одного персонажа
        </div>
      );
    }

    return (
      <div className="react-hw04__sw-counter" data-testid="with-loaded">
        Загружено персонажей: {loadedPeople}
      </div>
    );
  }
}

export { Counter, CounterProps };
