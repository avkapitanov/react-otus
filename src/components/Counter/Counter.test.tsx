import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Counter } from './Counter';

describe('Counter', () => {
  it('renders Empty component', () => {
    render(<Counter loadedPeople={0} />);

    const emptyCounter = screen.getByTestId('empty-counter');
    expect(emptyCounter).toBeInTheDocument();

    const empty = renderer.create(<Counter loadedPeople={0} />).toJSON();
    expect(empty).toMatchSnapshot();
  });

  it('renders with loaded elements component', () => {
    render(<Counter loadedPeople={5} />);

    const loadedCounter = screen.getByTestId('with-loaded');
    expect(loadedCounter).toBeInTheDocument();

    const loaded = renderer.create(<Counter loadedPeople={5} />).toJSON();
    expect(loaded).toMatchSnapshot();
  });
});
