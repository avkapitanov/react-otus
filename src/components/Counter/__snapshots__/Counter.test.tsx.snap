// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders Empty component 1`] = `
<div
  className="react-hw04__sw-counter react-hw04__sw-counter--empty"
  data-testid="empty-counter"
>
  Пока не загружено ни одного персонажа
</div>
`;

exports[`Counter renders with loaded elements component 1`] = `
<div
  className="react-hw04__sw-counter"
  data-testid="with-loaded"
>
  Загружено персонажей: 
  5
</div>
`;
